opensrp-fp/src/main/java/org/smartregister/fp/common/util/Utils.java

```diff
@@ -613,6 +619,44 @@ public static ButtonAlertStatus getButtonAlertStatus(Map<String, String> details
         return buttonAlertStatus;
     }
 
+    public static ButtonAlertStatus getButtonFollowupStatus(String triggerDate, ScheduleModel scheduleModel, String baseEntityId, String nextContactDate) {
+        ButtonAlertStatus buttonAlertStatus = new ButtonAlertStatus();
+
+        boolean isFirst = Utils.isUserFirstVisitForm(baseEntityId);
+        FPAlertRule fpAlertRule = new FPAlertRule(scheduleModel, triggerDate, isFirst);
+
+        buttonAlertStatus.buttonAlertStatus = FPLibrary.getInstance().getFPRulesEngineHelper()
+                .getFPAlertStatus(fpAlertRule, FP_ALERT_RULES);
+
+
+        buttonAlertStatus.nextContactDate = getFormatDateForVisitButton(nextContactDate);
+        return buttonAlertStatus;
+    }
+
+    public static String getFormatedDateForFPAlertRule(String triggerDate) {
+        SimpleDateFormat format = new SimpleDateFormat("dd-MM-yyyy");
+        Date newDate = null;
+        try {
+            newDate = format.parse(triggerDate);
+        } catch (ParseException e) {
+            e.printStackTrace();
+        }
+        format = new SimpleDateFormat("yyyy-MM-dd");
+        return format.format(newDate);
+    }
+
+    private static String getFormatDateForVisitButton(String triggerDate) {
```

Comment:
This method and getFormatedDateForFPAlertRule() appear to be doing the same thing, the only difference being the `triggerDate` date format and the `newDate` date format.
Is it possible to combine them into a generic function?